scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -466,6 +466,19 @@ class Router(formatOps: FormatOps) {
           Split(NewlineT(isDouble = tok.hasBlankLine), 0)
         )
 
+      case tok @ FormatToken(_: T.RightParen, _: T.KwDef | _: T.Comment, _)
```

Comment:
what happens if right is comment but after that we don't get a KwDef?

Reply:
That cannot happen, the code will not parse then.  Nothing significant can be added between `)` and `def`

Reply:
in that case, does it still make sense to match `KwDef` or `Comment` here? shouldn't `_` be equivalent, given the `leftOwner` check below?

Reply:
Ach, actually you can have a using clause `(using A)` and an opening bracket `{`  We want to format it differently in that case.

Reply:
If we leave it as just `_` it will format it a bit weirdly and also break the parser, since there is an issue when the formatting is a bit off. I will fix it separately, but the way it's currently formatted looks good to me.

Reply:
> Ach, actually you can have a using clause `(using A)` and an opening bracket `{` We want to format it differently in that case.

we can merge this as-is, but it would definitely be useful to have a follow-up pr to handle this:
```
extension [A, B](
  a: A
) /* foo */ (
  using b: B /* bar */
) /* baz */ {
  def a = 1
}
```